### DISCUSSION TITLE
Remove some elements from enum dropdown in Dialog.ts

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#6766] [Help] [answered]
I have a dropdown that is populated with values from an enum in a dialog.ts
I want to remove some elements from the dropdown depending on whether the user is creating a new record or is editing a record
Please assist

**Comment by edwardch** [comment, marked as answer]
I have a client table with an enum field called "State". The enum is called "ClientState".

Here we remove the "Completed" state from the quickfilter in the ClientGrid.ts file.

```ts
// ...
const fld = ClientRow.Fields;

// ...
protected myQuickFilter: EnumEditor;

// ...
protected createQuickFilters() {
    super.createQuickFilters();
    this.myQuickFilter = this.findQuickFilter(EnumEditor, fld.State);
    // remove one option from this.myQuickFilter
    this.myQuickFilter.items.splice(indexOf(this.myQuickFilter.items, x => x.id == String(ClientState.Completed)), 1);
}
```ts

HTH!

**Reply by rhemasystems** [reply]
Thanks, I had to make it `Q.indexOf` instead of `indexOf`
